app/scripts/modules/kubernetes/src/validation/applicationName.validator.js

```diff
@@ -8,17 +8,12 @@ module.exports = angular
   .module('spinnaker.kubernetes.validation.applicationName', [APPLICATION_NAME_VALIDATOR])
   .factory('kubernetesApplicationNameValidator', function() {
     function validateSpecialCharacters(name, warnings, errors) {
-      const alphanumPattern = /^([a-zA-Z][a-zA-Z0-9]*)?$/;
-      if (!alphanumPattern.test(name)) {
-        const alphanumWithDashPattern = /^([a-zA-Z][a-zA-Z0-9-]*)?$/;
-        if (alphanumWithDashPattern.test(name)) {
```

Comment:
Dashes are already allowed for Kubernetes, it just prints a helpful warning since https://github.com/spinnaker/deck/pull/4927